### PR TITLE
Update Dictionary Renderer Graphics Overlay sample to use new Ordered Anchor Points

### PR DESCRIPTION
**Pull request by shubham7109**
**Changes in this PR:**
- Converted sample from Java->Kotlin
- Removed Java sample
- Reverted back the sample to use the original XML file with `dictionarySymbolStyle` set to use "ORDERED ANCHOR POINT"
- Updated screenshot and README.md files. 
- Sample works using the XML file loaded using ADB

To check: 
- Not sure if the [data](https://github.com/Esri/arcgis-runtime-samples-android/tree/shu11698/update-dictionary-renderer-samples/kotlin/dictionary-renderer-with-graphics-overlay#about-the-data) used in the README file is correct. It is the URL as per the design doc. 
- Should we be should be using [this](https://github.com/Esri/arcgis-runtime-samples-ios/blob/90d12f8f15e5b3b02c744d0e262ad4f525517329/arcgis-ios-sdk-samples/Shared%20resources/Military%20messages/Mil2525DMessages.xml) XML file as our data source, which is hosted as a ZIP file [here](https://arcgisruntime.maps.arcgis.com/sharing/rest/content/items/623382e0113d40698538f249e3bcb1c0/data).
- The `Mil2525DMessages.xml` file is not being downloaded by the SampleViewer., most likely URL used in `provision_from` is not correct.

### CODE REVIEW

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Thanks @shubham7109! Mostly just need to sort out how the data's meant to be provisioned onto the device.

**Comment by XuFei1021** [comment]
你好，来信已收到，请放心。

File: kotlin/dictionary-renderer-with-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrendererwithgraphicsoverlay/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+        // create a map with the BasemapType topographic
+        val map = ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC)
+
+        // graphics no longer show after zooming passed this scale
+        graphicsOverlay.minScale = 1000000.0
+        mapView.graphicsOverlays.add(graphicsOverlay)
+
+        // create the dictionary symbol style from the Joint Military Symbology MIL-STD-2525D portal item
+        val portalItem = PortalItem(Portal("https://www.arcgis.com/", false), itemID)
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply kotlin/dictionary-renderer-with-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrendererwithgraphicsoverlay/MainActivity.kt:78]: Good idea, made the change 👍

File: kotlin/dictionary-renderer-with-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrendererwithgraphicsoverlay/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        val dictionarySymbolStyle = DictionarySymbolStyle(portalItem)
+
+        // add done loading listeners to the map and dictionary symbol style and check they have loaded
+        map.addDoneLoadingListener {
+            if (map.loadStatus == LoadStatus.LOADED) {
+                dictionarySymbolStyle.addDoneLoadingListener {
+                    if (dictionarySymbolStyle.loadStatus == LoadStatus.LOADED) {
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply kotlin/dictionary-renderer-with-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrendererwithgraphicsoverlay/MainActivity.kt:85]: Sure, I've added in the line, `if (dictionarySymbolStyle.loadStatus == LoadStatus.LOADED
                    && map.loadStatus == LoadStatus.LOADED)` to check for both the `map` and the `dictionarySymbolStyle`

File: kotlin/dictionary-renderer-with-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrendererwithgraphicsoverlay/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                        try {
+                            val messages: List<Map<String, Any>> = parseMessages()
+                            val graphics: MutableList<Graphic> = mutableListOf()
+                            // create graphics with attributes and add to graphics overlay
+                            messages.mapTo(graphics) { createGraphic(it) }
+                            graphicsOverlay.graphics.addAll(graphics)
+                            // set the viewpoint to the extent of the graphics overlay
+                            mapView.setViewpointGeometryAsync(graphicsOverlay.extent)
+                        } catch (e: Exception) {
+                            displayError("Error parsing messages: ${e.message}")
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply kotlin/dictionary-renderer-with-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrendererwithgraphicsoverlay/MainActivity.kt:104]: The function `displayError()` method does display the toast and the log-line. I added that just to reduce amount of lines used for logging. But I do agree with sticking to what we usually do. Added the changes

**Comment by shubham7109** [comment]
Changes made to fix provisioning error:
- Sample now downloads `Mil2525DMessages.xml` from [this](https://www.arcgis.com/home/item.html?id=623382e0113d40698538f249e3bcb1c0) AGOL item.
- Updated `README.metadata.json` provision URL 
- Updated `README.md` offline section to reflect corrected download link

Tested with the sample viewer, the sample builds and successfully provisions the `.xml` file

**Review by TADraeseke — APPROVED** [review]
Thanks @shubham7109 -- I see the DictionaryStyle's being created from an item now. Very cool.

Looks good! Just one small thought on the not loaded block below.

File: kotlin/dictionary-renderer-with-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrendererwithgraphicsoverlay/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                        mapView.setViewpointGeometryAsync(graphicsOverlay.extent)
+                    } catch (e: Exception) {
+                        val message = "Error parsing messages: ${e.message}"
+                        Log.e(TAG, message)
+                        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
+                    }
+                } else {
+                    val message = "Failed to load symbol style: ${dictionarySymbolStyle.loadError.cause?.message}"
+                    Log.e(TAG, message)
+                    Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply kotlin/dictionary-renderer-with-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrendererwithgraphicsoverlay/MainActivity.kt:108]: You're right, the map should be loaded for Dictionary style to load. Made the suggested change 👍

**Pull request merged**